agent/consul/state/connect_ca_test.go

```diff
@@ -59,8 +59,8 @@ func TestStore_CAConfigCAS(t *testing.T) {
 	ok, err := s.CACheckAndSetConfig(2, 0, &structs.CAConfiguration{
 		Provider: "static",
 	})
-	if ok || err != nil {
-		t.Fatalf("expected (false, nil), got: (%v, %#v)", ok, err)
+	if ok || err == nil {
+		t.Fatalf("expected (false, \"invalid index\"), got: (%v, %#v)", ok, err)
```

Comment:
I think it's always a good idea to at least check part of the error message. Otherwise when the code is changed the test could return an error for a different reason, and the code path that you expected to be testing is no longer covered.

```suggestion
require.False(t, ok)
testutil.RequireErrorContains(t, err, "ModifyIndex did not match existing")
```